src/js/map/TerrainGrid.js

```diff
@@ -0,0 +1,142 @@
+import Grid from './Grid'
+
+export default class TerrainGrid extends Grid {
+  constructor(tilesWide, tilesHigh) {
+    super(tilesWide, tilesHigh, 1)
+    this.initialize()
+  }
+
+  initialize() {
+    this.terrainProperties = {
+      normal: {
+        difficulty: 1,
+      },
+      crater: {
+        difficulty: 3,
+      },
+      ridge: {
+        difficulty: 2,
+      },
+      obstacle: {
+        difficulty: 0,
+      },
+    }
+    this.initialValue = this.getTerrainProperties("normal")
+    this.reset()
+  }
+
+  reset() {
+    super.reset()
+    this.generated = false;
+  }
+
+  getTerrainProperties(type) {
+    if (this.terrainProperties.hasOwnProperty(type)) {
+      return {
+        type: type,
+        difficulty: this.terrainProperties[type].difficulty,
+      }
+    }
+    return {
+      type: "normal",
+      difficulty: this.terrainProperties.normal.difficulty,
+    }
+  }
+
+  generateTerrain() {
+    if (this.generated) { return }
+    // Minimum crater size 1, maximum 3
+    const totalGridSpaces = this.tilesWide * this.tilesHigh
+    const defaultTerrainCoverage = 0.2
+    const targetTerrainCoverage = totalGridSpaces * defaultTerrainCoverage
+    let terrainCoverage = 0
+
+    do {
+      terrainCoverage +=
+          this.addTestCrater(this.getRandomGridLocation(),
+                             Math.floor(3 * Math.random()) + 1)
+    } while (terrainCoverage < targetTerrainCoverage)
+    this.generated = true;
+  }
+
+  // @TERRAIN
+  setTerrain(terrainData) {
+    // @TODO take terrain data and jam 'er in
+    this.values = terrainData
+  }
+
+  addTestCrater(gridLocation, size) {
```

Comment:
If we are using this for now, the 'test' part of the name should probably be removed!

Reply:
Done